python/ray/tests/test_autoscaler_yaml.py

```diff
@@ -25,6 +27,31 @@ def testValidateDefaultConfig(self):
             except Exception:
                 self.fail("Config did not pass validation test!")
 
+    @parameterized.expand([
+        ["tests/project_files/yaml_validation/additional_property.yaml"],
+        ["autoscaler/aws/development-example.yaml"],
+        ["autoscaler/aws/example-full.yaml"],
+        ["autoscaler/aws/example-gpu-docker.yaml"],
+        ["autoscaler/aws/example-minimal.yaml"],
+        ["autoscaler/gcp/example-full.yaml"],
+        ["autoscaler/gcp/example-gpu-docker.yaml"],
+        ["autoscaler/gcp/example-minimal.yaml"],
+        ["autoscaler/kubernetes/example-full.yaml"],
+        ["autoscaler/kubernetes/example-minimal.yaml"],
+        ["autoscaler/local/development-example.yaml"],
+        ["autoscaler/local/example-full.yaml"],
+    ])
+    def testInvalidConfig(self, config_path):
+        config = yaml.safe_load(os.path.join(RAY_PATH, config_path))
+
+        with open(os.path.join(os.path.dirname(ray.autoscaler.__file__), "ray-schema.json")) as f:
+            schema = f.read() 
+
+        try:
+            jsonschema.validate(config, json.loads(schema))
+            self.fail("Expected validation to fail")
+        except:
+            pass
```

Comment:
Instead of this, can you make a helper method `_test_invalid_config`, and then create separate tests for each set of yamls?

`testInvalidConfigKubernetes` (here, call `_test_invalid_config` on "autoscaler/kubernetes/example-full.yaml", "autoscaler/kubernetes/example-minimal.yaml"),

`testInvalidConfigLocal` (here, call `_test_invalid_config` on "autoscaler/local/example-full.yaml", "autoscaler/local/example-minimal.yaml"),

`testInvalidConfigGCP` (here, call `_test_invalid_config` on "autoscaler/gcp/example-full.yaml", "autoscaler/gcp/example-minimal.yaml"),

I would like to avoid introducing a new dependency (`parameterized`). After all, the test output doesn't actually seem that easy to decipher  `https://pypi.org/project/parameterized/`

Reply:
sure can change this. I hoped that in case of failure it's a bit easier to figure as it lists the config file.

Reply:
Yeah... i'm not sure (from the examples, it doesn't seem to be the case)